src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs

```diff
@@ -97,5 +99,19 @@ private void SetMessageField()
         }
 
         public virtual string? ParamName => _paramName;
+
+        /// <summary>Throws an <see cref="ArgumentNullException"/> if <paramref name="argument"/> is null. Throws an <see cref="ArgumentException"/> if <paramref name="argument"/> is an empty string.</summary>
+        /// <param name="argument">The reference type argument to validate as non-null and non-empty.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        public static void ThrowIfNullOrEmpty([NotNull] string? argument, [CallerArgumentExpression("argument")] string? paramName = null)
+        {
+            ArgumentNullException.ThrowIfNull(argument);
+
+            // Same style as written in string.IsNullOrEmpty
+            if (0 == argument.Length)
+            {
+                throw new ArgumentException(paramName);
```

Comment:
The first argument to ArgumentException is a message rather than a parameter name.  This should take both the resource string to use and the parameter name, and be extracted to a separate ThrowArgumentException method.

Reply:
@stephentoub  Thank you so much. Sorry for rookie mistakes. This is my first pull request here. Is there any test I should know/add to as well?